src/qibolab/platform.py

```diff
@@ -356,7 +354,30 @@ def execute_pulse_sequence(self, sequence, options, **kwargs):
                     result = new_result
         return result
 
-    def sweep(self, sequence, options, *sweepers):
+    def execute_pulse_sequences(self, sequences: List[PulseSequence], options: ExecutionParameters, **kwargs):
```

Comment:
Now there is some code repetition between this method and `execute_pulse_sequence`. If we lift the unrolling from the drivers (instruments) to the platform, as it was the initial idea, this will be fixed, so to me it is fine to leave as it is. 

But an easy solution would be to rename the existing `execute_pulse_sequence` to something like
```py
def _execute(self, method, sequences, options, **kwargs):
    ...
        new_result = getattr(instrument, method)(self.qubits, sequences, options)
    ...
```
and then call it from the other methods
```py
def execute_pulse_sequence(self, sequences: PulseSequence, options: ExecutionParameters, **kwargs):
    return self._execute("play", sequences, options, **kwargs)

def execute_pulse_sequences(self, sequences: List[PulseSequence], options: ExecutionParameters, **kwargs):
    return self._execute("play_sequences", sequences, options, **kwargs)
```

Reply:
Thanks for this, looks way better.